### PR TITLE
added deprecation to map.size property

### LINKED ISSUE #4337: Deprecate map.size
Given the discussion in #4202, we should deprecate the `size` attribute on `Map` for consistency as we do not want to add a `shape` attribute for the reasons listed in that issue.

### PR DESCRIPTION
**Pull request by dipanshu231099**
### Description

Added the required deprecation for the map.size property to use map.data.shape instead.

Fixes #4337

### CODE REVIEW

**Comment by nabobalis** [comment]
You will need to update:

```
FAILED ..\..\.tox\py37\lib\site-packages\sunpy\map\tests\test_mapbase.py::test_size - sunpy.util.exceptions.SunpyDeprecationWarning: Use map.data.shape instead
```
to catch and test for that exception.

**Comment by dstansby** [comment]
I don't think we should backport deprecations, so I've removed the label. @dipanshu231099 let us know if you need any help updating the tests!

**Comment by nabobalis** [comment]
> I don't think we should backport deprecations, so I've removed the label. @dipanshu231099 let us know if you need any help updating the tests!

I think we should so any people on the LTS will be aware of it and since it does have an alternative. If it did not have one, I would be against it.

**Comment by Cadair** [comment]
We wont be able to remove it until 3.1 but I don't think we should backport deprecations. Things which don't make the cut have to wait.

**Comment by nabobalis** [comment]
Ok, does this get a breaking changelog?

**Comment by Cadair** [comment]
No deprecations and removal of deprecated features go in `removal` according to the changelog readme.

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -531,6 +531,7 @@ def dtype(self):
         return self.data.dtype
 
     @property
+    @deprecated(since="2.1", message="Use map.data.shape instead", alternative="map.data.shape")
```
Comment (Cadair) [inline sunpy/map/mapbase.py:534]:
```suggestion
    @deprecated(since="2.1", message="Use map.data.size instead", alternative="map.data.size")
```

Comment (dipanshu231099) [reply sunpy/map/mapbase.py:534]: I guess the issue was to use map.data.shape instead of map.size property? or is it map.data.size?

**Comment by dipanshu231099** [comment]
@nabobalis the test to catch exception has to be done for the map.size property right ?

**Comment by nabobalis** [comment]
> @nabobalis the test to catch exception has to be done for the map.size property right ?

Yes.

File: changelog/4338.removal.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+depreacted the use of `sunpy.map.Map.size` property
```
Comment (nabobalis) [inline changelog/4338.removal.rst:1]:
```suggestion
Depreacted the use of `sunpy.map.Map.size` property
```

File: changelog/4338.removal.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Depreacted the use of `sunpy.map.Map.size` property
```
Comment (nabobalis) [inline changelog/4338.removal.rst:1]:
```suggestion
Deprecated the use of `sunpy.map.Map.size` property
```

**Comment by nabobalis** [comment]
Thanks @dipanshu231099

**Pull request merged**